x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -56,4 +92,38 @@ public Query wrapFunctionQuery(ScalarFunction sf, Expression field, Supplier<Que
         }
         throw new EsqlIllegalArgumentException("Expected a FieldAttribute or MetadataAttribute but received [" + field + "]");
     }
+
+    public static class EqualsIgnoreCaseTranslator extends ExpressionTranslator<InsensitiveEquals> {
+
+        @Override
+        protected Query asQuery(InsensitiveEquals bc, TranslatorHandler handler) {
+            return doTranslate(bc, handler);
+        }
+
+        public static Query doTranslate(InsensitiveEquals bc, TranslatorHandler handler) {
+            checkInsensitiveComparison(bc);
+            return handler.wrapFunctionQuery(bc, bc.left(), () -> translate(bc));
+        }
+
+        public static void checkInsensitiveComparison(InsensitiveEquals bc) {
+            Check.isTrue(
+                bc.right().foldable(),
+                "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [{}]",
+                bc.right().sourceLocation().getLineNumber(),
+                bc.right().sourceLocation().getColumnNumber(),
+                Expressions.name(bc.right()),
+                bc.symbol()
+            );
+        }
+
+        static Query translate(InsensitiveEquals bc) {
+            TypedAttribute attribute = checkIsPushableAttribute(bc.left());
+            Source source = bc.source();
+            BytesRef value = BytesRefs.toBytesRef(ExpressionTranslators.valueOf(bc.right()));
+            String name = pushableAttributeName(attribute);
+            Query query = new WildcardQuery(source, name, value.utf8ToString(), true);
```

Comment:
This is incorrect - the equality should translate to a Term query with case_insensitive. The wildcard query is used only by LIKE.

Reply:
Maybe I got it completely wrong, but from the original [issue](https://github.com/elastic/elasticsearch/issues/103599) and from the off-line discussions my understanding was that we wanted wildcard expressions on the right-hand side. 
If we want it, then we'll need `WildcardQuery` here.